onnx/test/parser_test.py

```diff
@@ -218,6 +218,36 @@ def test_parse_node(self):
         self.assertEqual(node.domain, "SomeDomain")
         self.assertEqual(node.op_type, "SomeOp")
 
+    @parameterized.expand(
+        [
+            "not_a_good_float",  # Not a good float value, but shouldn't blow up the parser.
+            "inf",
+            "-inf",
+            "infinity",
+            "-infinity",
+            "-NaN",
+            "nan",
+        ]
+    )
+    def test_parse_various_float_values(self, test_literal):
+        model_text = (
+            """
+        <
+        ir_version: 8,
+        opset_import: ["" : 18, "onnxscript.atenlib" : 1, "this" : 1],
+        producer_name: "pytorch",
+        producer_version: "2.1.0"
+        >
+        _func () => ()
+        {
+        tmp_11 = Constant <value_float = """
+            + test_literal
+            + """> ()
+        }
+        """
+        )
+        onnx.parser.parse_model(model_text)
+
```

Comment:
The unit test should include some cases where inf is used a variable name, or it is mispelled just to make sure it raises an exception or works as expected in these cases.